compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1448,7 +1448,7 @@ object desugar {
               ))
         }
       case _ =>
-        EmptyTree //may happen for erroneous input
+        EmptyTree // may happen for erroneous input. An error will already have been reported.
```

Comment:
Instead of leaving it as a comment, you can enforce this with `assert(ctx.reporter.errorsReported)` (we do this in several places already)

Reply:
Done.